common/CommonAPI.h

```diff
@@ -204,90 +204,19 @@ class CommonAPI
 			Channels<nbl::ui::IKeyboardEventChannel> m_keyboard;
 	};
 
-	// TODO: can you guys just use one callback!?
 	class ICommonAPIEventCallback : public nbl::ui::IWindow::IEventCallback
 	{
 	public:
 		virtual void setLogger(nbl::system::logger_opt_smart_ptr& logger) = 0;
-	};
-	class CTemporaryEventCallback : public ICommonAPIEventCallback
-	{
-		nbl::system::logger_opt_smart_ptr m_logger = nullptr;
-	public:
-		void setLogger(nbl::system::logger_opt_smart_ptr& logger) override
-		{
-			m_logger = logger;
-		}
-	private:
-		bool onWindowShown_impl() override
-		{
-			m_logger.log("Window Shown");
-			return true;
-		}
-		bool onWindowHidden_impl() override
-		{
-			m_logger.log("Window hidden");
-			return true;
-		}
-		bool onWindowMoved_impl(int32_t x, int32_t y) override
-		{
-			m_logger.log("Window window moved to { %d, %d }", nbl::system::ILogger::ELL_WARNING, x, y);
-			return true;
-		}
-		bool onWindowResized_impl(uint32_t w, uint32_t h) override
-		{
-			m_logger.log("Window resized to { %u, %u }", nbl::system::ILogger::ELL_DEBUG, w, h);
-			return true;
-		}
-		bool onWindowMinimized_impl() override
-		{
-			m_logger.log("Window minimized", nbl::system::ILogger::ELL_ERROR);
-			return true;
-		}
-		bool onWindowMaximized_impl() override
-		{
-			m_logger.log("Window maximized", nbl::system::ILogger::ELL_PERFORMANCE);
-			return true;
-		}
-		void onGainedMouseFocus_impl() override
-		{
-			m_logger.log("Window gained mouse focus", nbl::system::ILogger::ELL_INFO);
-		}
-		void onLostMouseFocus_impl() override
-		{
-			m_logger.log("Window lost mouse focus", nbl::system::ILogger::ELL_INFO);
-		}
-		void onGainedKeyboardFocus_impl() override
-		{
-			m_logger.log("Window gained keyboard focus", nbl::system::ILogger::ELL_INFO);
-		}
-		void onLostKeyboardFocus_impl() override
-		{
-			m_logger.log("Window lost keyboard focus", nbl::system::ILogger::ELL_INFO);
-		}
-		void onMouseConnected_impl(nbl::core::smart_refctd_ptr<nbl::ui::IMouseEventChannel>&& mch) override
-		{
-			m_logger.log("A mouse has been connected", nbl::system::ILogger::ELL_INFO);
-		}
-		void onMouseDisconnected_impl(nbl::ui::IMouseEventChannel* mch) override
-		{
-			m_logger.log("A mouse has been disconnected", nbl::system::ILogger::ELL_INFO);
-		}
-		void onKeyboardConnected_impl(nbl::core::smart_refctd_ptr<nbl::ui::IKeyboardEventChannel>&& kbch) override
-		{
-			m_logger.log("A keyboard has been connected", nbl::system::ILogger::ELL_INFO);
-		}
-		void onKeyboardDisconnected_impl(nbl::ui::IKeyboardEventChannel* kbch) override
-		{
-			m_logger.log("A keyboard has been disconnected", nbl::system::ILogger::ELL_INFO);
-		}
 
-		bool onWindowClosed_impl() override
+		void setApplication(nbl::ui::IGraphicalApplicationFramework* app)
 		{
-			m_logger.log("Window closed");
-			return true;
+			m_app = std::move(app);
 		}
+	protected:
+		nbl::ui::IGraphicalApplicationFramework* m_app = nullptr;
 	};
+
 	class CommonAPIEventCallback : public ICommonAPIEventCallback
```

Comment:
why do you need this weird inheritance?

Reply:
I suggested to remove this in a PR comment as well